### PR TITLE
Add route styling example.

### PR DESCRIPTION
**Pull request by MaximAlien**
Closing #81.

This example showcases the ability to change main and alternative routes inside `NavigationMapView`.

One of possible examples:
Screen Shot 2020-11-10 at 2 36 12 PM

### CODE REVIEW

File: Navigation-Examples/Examples/Route-Styling.swift [hunk 1]
```diff
@@ -0,0 +1,350 @@
...
+        var shape: [MGLPolylineFeature] = []
+        
+        for leg in route.legs {
+            let coordinates: [CLLocationCoordinate2D] = leg.steps.enumerated().reduce([]) { allCoordinates, current in
+                let stepCoordinates = current.element.shape!.coordinates
+                
+                return current.offset == 0 ? stepCoordinates : allCoordinates + stepCoordinates.suffix(from: 1)
+            }
+            
+            if let congestionLevel = CongestionLevel.allCases.randomElement() {
```
Comment (MaximAlien) [inline Navigation-Examples/Examples/Route-Styling.swift:340]: This change requires slight update in Mapbox Directions: https://github.com/mapbox/mapbox-directions-swift/pull/500

File: Podfile [hunk 2]
```diff
@@ -1,7 +1,8 @@
 platform :ios, '10.0'
 use_frameworks!
 
-pod 'MapboxNavigation', '~> 1.0'
+pod 'MapboxCoreNavigation', :git => 'https://github.com/mapbox/mapbox-navigation-ios.git', :branch => 'maxim/651-route-line-styling'
```
Comment (MaximAlien) [inline Podfile:4]: This PR depends on changes from https://github.com/mapbox/mapbox-navigation-ios/pull/2719.

File: Navigation-Examples/Constants.swift [hunk 3]
```diff
@@ -97,5 +97,12 @@ let listOfExamples: [NamedController] = [
         controller: BuildingExtrusionViewController.self,
         storyboard: nil,
         pushExampleToViewController: true
+    ),
+    (
+        name: "Route Styling",
```
Comment (1ec5) [inline Navigation-Examples/Constants.swift:102]: “Route line styling” would be less prone to confusion with other kinds of styling that the examples demonstrate.

**Review by 1ec5 — APPROVED** [review]
I’m assuming the example looks similar to the screen recording in https://github.com/mapbox/mapbox-navigation-ios/pull/2719#issue-513320202. The code is fairly involved for an example, but it makes sense that a lot needs to be set up in order to configure route line styling.

**Pull request merged**